server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/SpamAssassin.java

```diff
@@ -111,8 +111,8 @@ private void querySpamAssassin(Mail mail, MimeMessage message, SpamAssassinInvok
         // Add headers per recipient to mail object
         for (String key : result.getHeadersAsAttribute().keySet()) {
             mail.addSpecificHeaderForRecipient(PerRecipientHeaders.Header.builder()
-                    .name(key)
-                    .value(result.getHeadersAsAttribute().get(key))
+                    .name(key.asString())
+                    .value((String) result.getHeadersAsAttribute().get(key).getValue().getValue())
```

Comment:
How about adding to AttributeValue a `<T> getValue(Class<T> type);` ?

Reply:
It looks more acceptable.

Reply:
Well, thinking a bit about it I think it's a non-sense, you have an `Attribute<T>`, a cast to a type you are supposed to know is error-prone at best.